examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_friction.py

```diff
@@ -0,0 +1,279 @@
+                self.callback_params["lengths"].append(
+                    cylinder_length_collection.copy()
+                )
+                self.callback_params["com_velocity"].append(
+                    system.velocity_collection[..., 0].copy()
+                )
+
+                total_energy = (
+                    system.compute_translational_energy()
+                    + system.compute_rotational_energy()
+                )
+                self.callback_params["total_energy"].append(total_energy[..., 0].copy())
+
+                return
+
+    if POST_PROCESSING:
+        post_processing_dict_list.append(defaultdict(list))
+        rod_cylinder_parallel_contact_simulator.collect_diagnostics(rod).using(
+            StraightRodCallBack,
+            step_skip=step_skip,
+            callback_params=post_processing_dict_list[0],
+        )
+        # For rigid body
+        post_processing_dict_list.append(defaultdict(list))
+        rod_cylinder_parallel_contact_simulator.collect_diagnostics(rigid_body).using(
+            RigidCylinderCallBack,
+            step_skip=step_skip,
+            callback_params=post_processing_dict_list[1],
+            resize_cylinder_elems=n_elem,
+        )
+
+    rod_cylinder_parallel_contact_simulator.finalize()
+    timestepper = PositionVerlet()
+
+    integrate(
+        timestepper, rod_cylinder_parallel_contact_simulator, final_time, total_steps
+    )
+
+    if POST_PROCESSING:
+        # Plot the rods
+        plot_video_with_surface(
+            post_processing_dict_list,
+            video_name="rod_cylinder_contact.mp4",
+            fps=rendering_fps,
+            step=1,
+            # The following parameters are optional
+            x_limits=(-base_length * 5, base_length * 5),  # Set bounds on x-axis
+            y_limits=(-base_length * 5, base_length * 5),  # Set bounds on y-axis
+            z_limits=(-base_length * 5, base_length * 5),  # Set bounds on z-axis
+            dpi=100,  # Set the quality of the image
+            vis3D=True,  # Turn on 3D visualization
+            vis2D=True,  # Turn on projected (2D) visualization
+        )
+
+        filaname = "rod_rigid_velocity.png"
+        plot_velocity(
+            post_processing_dict_list[0],
+            post_processing_dict_list[1],
+            filename=filaname,
+            SAVE_FIGURE=True,
+        )
+
+    # Compute final total energy
+    total_final_energy = (
+        rod.compute_translational_energy()
+        + rod.compute_rotational_energy()
+        + rod.compute_bending_energy()
+        + rod.compute_shear_energy()
+    )
+
+    return total_final_energy
+
+
+if __name__ == "__main__":
+    import multiprocessing as mp
+    from examples.RigidBodyCases.RodRigidBodyContact.post_processing import (
+        plot_force_vs_energy,
+    )
+
+    # total_energy = rod_cylinder_contact_friction_case(pulling_force=0.5, POST_PROCESSING=True)
+
+    force = list([(float(x)) / 100.0 for x in range(0, 90, 5)])
```

Comment:
here we can then fix normal_force and iterate over pulling_force = gamma * normal_force with gamma in a range from 0 to 1 ?